### PR TITLE
Allow for custom results formatters

### PR DESCRIPTION
**Pull request by jwsloan**
I've [started working on](https://github.com/jwsloan/codeclimate-repolinter) a [CodeClimate](https://codeclimate.com/) [Engine](https://docs.codeclimate.com/docs/building-a-code-climate-engine) that uses `repolinter`. To complete it, I need to have the output adhere to Code Climate's [issue](https://github.com/codeclimate/spec/blob/master/SPEC.md#issues) spec.
This change will allow me (or anyone else) to do that, and also allow JSON output as request in issue #43.

### CODE REVIEW

File: formatters/json_formatter.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 TODO Group. All rights reserved.
+// Licensed under the Apache License, Version 2.0.
+
+class JsonFormatter {
+  format (rule, message, level) {
+    let output = {}
```
Comment (sschuberth) [inline formatters/json_formatter.js]: How about inlining the initialization like

    let output = {
        rule: rule.id,
        ...

?

Comment (jwsloan) [reply formatters/json_formatter.js]: Good call. Will do.

Comment (jwsloan) [reply formatters/json_formatter.js]: Thanks for the quick response, @sschuberth! I've made that change.

**Comment by caniszczyk** [comment]
Awesome @jwsloan , thanks!!!

**Comment by jwsloan** [comment]
You're welcome, @caniszczyk! Thanks for merging!

**Pull request merged**